### DISCUSSION TITLE
https://1kevinson.com/how-to-build-rest-api-with-spring-boot-and-postgresql/

### DISCUSSION THREAD

**Discussion by giscus** [1kevinson/BLOG-TUTORIALS#9] [Announcements]
# https://1kevinson.com/how-to-build-rest-api-with-spring-boot-and-postgresql/

In this tutorial you will learn how to build a complete Rest API with Spring Boot and PostgresSQL 15

https://1kevinson.com/how-to-build-rest-api-with-spring-boot-and-postgresql/

**Comment by deltadelta** [comment]
Hey Kevin - thanks for the tutorial, it's really helpful for the project I'm trying get off the ground.

I'm having a bit of an issue when it comes to building the docker image, and I'm not sure if it's because the openjdk:alpine19 image you listed is deprecated or not.

```
./mvnw clean package -DskipTest &&
docker compose build --no-cache &&
docker compose up -d
```

Is returning

```
[+] Building 0.8s (3/3) FINISHED                                                                               
 => [internal] load build definition from spring.dockerfile                                               0.0s
 => => transferring dockerfile: 235B                                                                      0.0s
 => [internal] load .dockerignore                                                                         0.0s
 => => transferring context: 2B                                                                           0.0s
 => ERROR [internal] load metadata for docker.io/library/openjdk:19-alpine                                0.7s
------
 > [internal] load metadata for docker.io/library/openjdk:19-alpine:
------
failed to solve: rpc error: code = Unknown desc = failed to solve with frontend dockerfile.v0: failed to create LLB definition: rpc error: code = Unknown desc = error getting credentials - err: exit status 1, out:
```


Any help here would be much appreciate, thank you!

**Reply by 1kevinson** [reply]
Hello Diego,

I didn't encounter this problem during my local development, but there are some people who apparently faced the same problem online, 

I suggest you to try one of these 3 methods: 
- Change the name of your docker file and start it with capital D. example: ```spring.Dockerfile``` or ```Dockerfile.spring```
- Update your Docker application on your workstation
- Restart your Docker application on your workstation

Here are links to more information about your problem:
- https://stackoverflow.com/questions/64221861/an-error-failed-to-solve-with-frontend-dockerfile-v0
- https://stackoverflow.com/questions/64985913/failed-to-solve-with-frontend-dockerfile

I hope I helped you mate

**Reply by deltadelta** [reply]
Hey, thank you for the response.

It appears that since I am running on an Apple Silicon Machine, which cannot natively run x86 images, and that there is no arm64 version of the image - docker was unable to download it.

I'm going to setup a remote development environment running x86 and see if that solves my issue.

Thanks again - Diego

**Reply by 1kevinson** [reply]
It's weird, I'm using an Apple Silicon M2 Mac Mini, and I have not faced this issue.

**Comment by entelect-sorlim** [comment]
Hi,

I followed this (am new to java, but old to programming) and well.... 

15/02/2024 11:05 - Build completed with 47 errors and 0 warnings in 2 sec, 93 ms

There's so much that isn't covered here - doesn't seem like all the necessary dependencies are in place for example.

Have you tried following this guide verbatim? I have all the code in place and did everything that is stated - but many things must be missing.

Take care

**Reply by 1kevinson** [reply]
Hi @etelect-sorlim, 

I tried the tutorial myself, and everything works well. you can find the full source code [here](https://github.com/1kevinson/BLOG-TUTORIALS/tree/master/Java/spring-boot-rest-api?ref=1kevinson.com) on my github. :)
